### ISSUE TITLE
UserTiming.mark is showing up in traces

### ISSUE THREAD

**Issue by natduca** [catapult-project/catapult#1950]
@lpy @zeptonaut I'm seeing marks showing up as ThreadSlices in the model. Is this intentional? Is it "sane"?

**Comment by natduca** [comment]
![image](https://cloud.githubusercontent.com/assets/412396/12744920/beec1368-c94b-11e5-8ee7-f92cb397544f.png)

Those little dashes are the marks... its pretty strange.

**Comment by zeptonaut** [comment]
Ack - yep, this is my fault. It seems like we want something that's process-level, but doesn't go into the current nesting hierarchy. I see that there's something that's called a `ProcessInstantEvent`. Do you know if that gets rendered in a way that's sensible for what we want here?

**Comment by lpy** [comment]
Is it because of some changes in chromium? @zeptonaut

**Comment by natduca** [comment]
These all get rendered pretty poorly by chrome. I think the basic question is "how should marks show up in traces? Should they show up?

@paulirish can you help?

**Comment by paulirish** [comment]
They definitely should be in the file. When measures landed in devtools, the lack of marks came up very quick: https://twitter.com/igrigorik/status/690636030727159808

I would expect marks would get treatment on a solo track. IE has a nice representation of this:
![image](https://cloud.githubusercontent.com/assets/39191/13368300/0e7fa498-dc9e-11e5-81eb-6b79082d5291.png) (from [Performance : Microsoft Edge Dev](https://dev.windows.com/en-us/microsoft-edge/platform/documentation/f12-devtools-guide/performance/))

This track would ideally be by the ruler and essentially position-fixed. 

> if i have a mark that is never referenced in a measure, should we display that as a big vertical bar, but then when i have mark(a), mark(b), measure(c, a, b), then maybe we shouldn't show the marks for 'a' and 'b' but should show the measure 'c'.

Yup exactly. Clicking on that measure could reveal the names of a and b. But they needed be visualized as independent items. 

@igrigorik that all sound good?

**Comment by igrigorik** [comment]
sgtm!

**Comment by zeptonaut** [comment]
I'm not sure I agree with the idea that they should be position-fixed near the ruler. That makes a lot of sense in devtools, where we're only ever showing data for one page at a time. In trace viewer, though, we can be showing data for lots of pages, and having all of those pages' marks jumbled up in a single track near the ruler sounds confusing. I'd be a proponent of having them as the top track under each tab's renderer so that the marks/measures are separated out..

**Comment by benshayden** [comment]
The relationship between tabs, frames, and renderers is complex, so maybe, in the post-frameblamr world, we might want a binning track per (frame/tab) context, with marks at the top of that?

**Comment by natduca** [comment]
It sounds like the simple version of this is that we take instant events
and make them use the LetterDotTrack instead of havng them draw big
vertical bars. That doesn't rely on future tech like frames & tabs, but
lets us have these in the trace but less painfully.

Main question is if these should be process wide instant events or
thread-scoped instant events. It looks like they're currently thread
scoped. I am thinking maybe thats' not right?

**Comment by zeptonaut** [comment]
Yep, I agree. It seems like it makes more sense for them to be process-based.

**Comment by zeptonaut** [comment]
@natduca What about measures? This is how they currently look.

![measure](https://cloud.githubusercontent.com/assets/1407397/13466418/d7b34c30-e067-11e5-9ce6-fe253989b672.png)

It seems pretty unlikely that this is the experience that we want, but I'm not too sure of how else we'd do it.

We could put them all on the same track, but what if some of them overlap? Right now, there's only one async event happening at a time per track, IIUC.

Frankly, it's probably not *quite* as important that we get the experience perfect for measures. I It sounds like people rely on marks more heavily.

**Comment by zeptonaut** [comment]
Taking over this one, if that's alright @lpy. I already have a solution whipped up that wasn't too bad.

**Comment by lpy** [comment]
For measure, currently we use group name to group them, one group name one track. Making all into one track may be confusing.

**Comment by natduca** [comment]
I think measures are shown about how we'd like. I could imagine just taking
all the marks and showing them all in one place. I'd welcome a vc with the
folks active on this thread!
